### ISSUE TITLE
Discriminator LR Decay

### ISSUE THREAD

**Issue by normster** [szq0214/MEAL-V2#9]
Thanks for your work and the code release!

I have a small question about the lr decay schedule for the discriminator- the initial lr value for the discriminator is set to 1e-4 but it looks like it gets clobbered with the student lr value in `_set_learning_rate`:

https://github.com/szq0214/MEAL-V2/blob/3558f37175f2a9e0514eb013a2021d344ef612b1/train.py#L94-L96

Is this intentional? The discriminator is a simple model so I don't think this would make a big difference either way.

Thanks

**Comment by szq0214** [comment]
Hi @normster, thanks very much for pointing out this and it is not intentional. I also have noticed this small bug recently but since our models are trained with the currently released codebase and I'm tied up with some deadlines, I may fix it later and see what happens. I totally agree with you this would not make a big difference.

**Comment by normster** [comment]
Thanks! I realized while re-implementing this that if you adjust the learning rate every epoch this does cause training to diverge (e.g. using cos lr decay or some implementations of waterfall lr decay) and you do in fact need to start with a low initial lr for the discriminator. 

Also, I might have missed this from your paper but did you run any ablations without the discriminator? It's quite counterintuitive to me that distinct student/teacher logits are a good thing since if your student could perfectly mirror your teacher that would "solve" distillation.

**Comment by szq0214** [comment]
Hi @normster, I think it's a trade-off, perfectly mimicking the teacher's output is not a good thing, the student may just memorize all the distributions of training samples and the generalization is not guaranteed. Also, it is difficult for the student to completely mirror the teacher since the scale of the student is much smaller than the teacher. We have some preliminary comparison and we did not see any negative effect with the discriminator. It is not counterintuitive as it can be regarded as the regularization.

**Comment by szq0214** [comment]
@normster, below is the comparison of Top-1 accuracy curves on the ImageNet val set with and without the discriminator.

![test_top1_w_wo_dis](https://user-images.githubusercontent.com/3794909/96163224-9ed62280-0ee7-11eb-91f3-b5eed541af1d.png)

**Comment by zimenglan-sysu-512** [comment]
hi @szq0214 
so the discriminator is not important?
